### ISSUE TITLE
weave spamming:  Cannot forward frame yet - awaiting contact

### ISSUE THREAD

**Issue by bmurphy1976** [zettio/weave#404]
On at least two servers I've got weave 0.9.0 spamming Cannot forward frame yet - awaiting contact.  I can't figure out why or how to stop it.

```
weave 2015/02/18 21:46:15.166429 ->[7a:05:6a:8c:b1:3f]: Cannot forward frame yet - awaiting contact
weave 2015/02/18 21:46:15.961735 ->[7a:27:ac:82:83:6a]: Cannot forward frame yet - awaiting contact
weave 2015/02/18 21:46:15.962064 ->[7a:05:6a:8c:b1:3f]: Cannot forward frame yet - awaiting contact
weave 2015/02/18 21:46:16.165546 ->[7a:27:ac:82:83:6a]: Cannot forward frame yet - awaiting contact
weave 2015/02/18 21:46:16.165786 ->[7a:05:6a:8c:b1:3f]: Cannot forward frame yet - awaiting contact
weave 2015/02/18 21:46:16.166102 ->[7a:27:ac:82:83:6a]: Cannot forward frame yet - awaiting contact
weave 2015/02/18 21:46:16.166336 ->[7a:05:6a:8c:b1:3f]: Cannot forward frame yet - awaiting contact
weave 2015/02/18 21:46:16.961517 ->[7a:27:ac:82:83:6a]: Cannot forward frame yet - awaiting contact
weave 2015/02/18 21:46:16.961540 ->[7a:05:6a:8c:b1:3f]: Cannot forward frame yet - awaiting contact
weave 2015/02/18 21:46:17.165717 ->[7a:27:ac:82:83:6a]: Cannot forward frame yet - awaiting contact
weave 2015/02/18 21:46:17.165741 ->[7a:05:6a:8c:b1:3f]: Cannot forward frame yet - awaiting contact
weave 2015/02/18 21:46:17.964439 ->[7a:27:ac:82:83:6a]: Cannot forward frame yet - awaiting contact
weave 2015/02/18 21:46:17.964686 ->[7a:05:6a:8c:b1:3f]: Cannot forward frame yet - awaiting contact
weave 2015/02/18 21:46:17.965087 ->[7a:27:ac:82:83:6a]: Cannot forward frame yet - awaiting contact
weave 2015/02/18 21:46:17.965390 ->[7a:05:6a:8c:b1:3f]: Cannot forward frame yet - awaiting contact
weave 2015/02/18 21:46:18.165601 ->[7a:27:ac:82:83:6a]: Cannot forward frame yet - awaiting contact
weave 2015/02/18 21:46:18.165883 ->[7a:05:6a:8c:b1:3f]: Cannot forward frame yet - awaiting contact
weave 2015/02/18 21:46:18.961494 ->[7a:27:ac:82:83:6a]: Cannot forward frame yet - awaiting contact
weave 2015/02/18 21:46:18.961752 ->[7a:05:6a:8c:b1:3f]: Cannot forward frame yet - awaiting contact
weave 2015/02/18 21:46:18.962190 ->[7a:27:ac:82:83:6a]: Cannot forward frame yet - awaiting contact
weave 2015/02/18 21:46:18.962274 ->[7a:05:6a:8c:b1:3f]: Cannot forward frame yet - awaiting contact
```

Here's weave status.  We only have about 7 servers running 20-30 containers in this cluster.

```
# weave status
weave router 0.9.0
Encryption on
Our name is 7a:76:1a:fe:d2:3c
Sniffing traffic on &{112203 65535 ethwe ca:cc:47:a1:fa:7b up|broadcast|multicast}
MACs:
02:42:c0:a8:04:34 -> 7a:c3:ea:78:72:dc (2015-02-18 21:40:07.249381189 +0000 UTC)
02:42:c0:a8:04:3a -> 7a:c3:ea:78:72:dc (2015-02-18 21:40:08.036529412 +0000 UTC)
02:42:c0:a8:01:11 -> 7a:76:1a:fe:d2:3c (2015-02-18 21:47:55.836742824 +0000 UTC)
02:42:c0:a8:0e:01 -> 7a:27:ac:82:83:6a (2015-02-18 21:48:45.501869512 +0000 UTC)
02:42:c0:a8:04:3b -> 7a:c3:ea:78:72:dc (2015-02-18 21:40:08.218323579 +0000 UTC)
02:42:c0:a8:04:3c -> 7a:c3:ea:78:72:dc (2015-02-18 21:40:08.485655531 +0000 UTC)
02:42:c0:a8:04:39 -> 7a:c3:ea:78:72:dc (2015-02-18 21:40:08.279755288 +0000 UTC)
02:42:c0:a8:04:54 -> 7a:c3:ea:78:72:dc (2015-02-18 21:47:04.620907277 +0000 UTC)
02:42:c0:a8:04:56 -> 7a:c3:ea:78:72:dc (2015-02-18 21:48:04.012375429 +0000 UTC)
02:42:c0:a8:04:35 -> 7a:c3:ea:78:72:dc (2015-02-18 21:40:07.380217507 +0000 UTC)
02:42:c0:a8:04:3f -> 7a:c3:ea:78:72:dc (2015-02-18 21:41:04.542017611 +0000 UTC)
02:42:c0:a8:01:02 -> 7a:76:1a:fe:d2:3c (2015-02-18 21:48:08.855977674 +0000 UTC)
7a:c3:ea:78:72:dc -> 7a:c3:ea:78:72:dc (2015-02-18 21:48:12.748212899 +0000 UTC)
02:42:c0:a8:04:4a -> 7a:c3:ea:78:72:dc (2015-02-18 21:45:05.510528032 +0000 UTC)
02:42:c0:a8:04:55 -> 7a:c3:ea:78:72:dc (2015-02-18 21:48:03.947155982 +0000 UTC)
02:42:c0:a8:10:01 -> 7a:23:07:a5:99:a2 (2015-02-18 21:48:51.441854171 +0000 UTC)
7a:76:1a:fe:d2:3c -> 7a:76:1a:fe:d2:3c (2015-02-18 21:48:27.981537461 +0000 UTC)
02:42:c0:a8:04:53 -> 7a:c3:ea:78:72:dc (2015-02-18 21:47:04.570520428 +0000 UTC)
02:42:c0:a8:04:57 -> 7a:c3:ea:78:72:dc (2015-02-18 21:48:04.364913507 +0000 UTC)
02:42:c0:a8:04:37 -> 7a:c3:ea:78:72:dc (2015-02-18 21:40:07.959461622 +0000 UTC)
02:42:c0:a8:04:3d -> 7a:c3:ea:78:72:dc (2015-02-18 21:41:04.457768182 +0000 UTC)
02:42:c0:a8:04:40 -> 7a:c3:ea:78:72:dc (2015-02-18 21:42:03.450098412 +0000 UTC)
02:42:c0:a8:04:4e -> 7a:c3:ea:78:72:dc (2015-02-18 21:45:06.088000214 +0000 UTC)
02:42:c0:a8:04:4d -> 7a:c3:ea:78:72:dc (2015-02-18 21:45:06.133386945 +0000 UTC)
02:42:c0:a8:04:36 -> 7a:c3:ea:78:72:dc (2015-02-18 21:40:07.569382088 +0000 UTC)
7a:49:7e:1a:50:df -> 7a:49:7e:1a:50:df (2015-02-18 21:48:09.884530307 +0000 UTC)
02:42:c0:a8:04:43 -> 7a:c3:ea:78:72:dc (2015-02-18 21:43:03.759819147 +0000 UTC)
02:42:c0:a8:04:52 -> 7a:c3:ea:78:72:dc (2015-02-18 21:47:04.046963945 +0000 UTC)
02:42:c0:a8:04:47 -> 7a:c3:ea:78:72:dc (2015-02-18 21:44:04.430967288 +0000 UTC)
02:42:c0:a8:04:49 -> 7a:c3:ea:78:72:dc (2015-02-18 21:45:05.501378218 +0000 UTC)
7a:05:6a:8c:b1:3f -> 7a:05:6a:8c:b1:3f (2015-02-18 21:40:59.945450891 +0000 UTC)
02:42:c0:a8:04:0c -> 7a:c3:ea:78:72:dc (2015-02-18 21:48:48.592143918 +0000 UTC)
02:42:c0:a8:16:01 -> 7a:05:6a:8c:b1:3f (2015-02-18 21:48:10.505662462 +0000 UTC)
02:42:c0:a8:04:4c -> 7a:c3:ea:78:72:dc (2015-02-18 21:45:05.942477802 +0000 UTC)
02:42:c0:a8:04:38 -> 7a:c3:ea:78:72:dc (2015-02-18 21:40:07.641461163 +0000 UTC)
02:42:c0:a8:04:45 -> 7a:c3:ea:78:72:dc (2015-02-18 21:43:04.318427742 +0000 UTC)
02:42:c0:a8:04:41 -> 7a:c3:ea:78:72:dc (2015-02-18 21:42:03.669773462 +0000 UTC)
02:42:c0:a8:04:48 -> 7a:c3:ea:78:72:dc (2015-02-18 21:44:04.498693517 +0000 UTC)
02:42:c0:a8:04:44 -> 7a:c3:ea:78:72:dc (2015-02-18 21:43:04.037384324 +0000 UTC)
02:42:c0:a8:04:46 -> 7a:c3:ea:78:72:dc (2015-02-18 21:44:04.289988057 +0000 UTC)
02:42:c0:a8:04:4b -> 7a:c3:ea:78:72:dc (2015-02-18 21:45:05.699232698 +0000 UTC)
02:42:c0:a8:0f:02 -> 7a:71:93:51:24:98 (2015-02-18 21:47:53.627641262 +0000 UTC)
02:42:c0:a8:04:42 -> 7a:c3:ea:78:72:dc (2015-02-18 21:42:03.8683145 +0000 UTC)
02:42:c0:a8:04:3e -> 7a:c3:ea:78:72:dc (2015-02-18 21:41:04.415810672 +0000 UTC)
Peers:
Peer 7a:49:7e:1a:50:df (v7924) (UID 13094787421077564308)
   -> 7a:c3:ea:78:72:dc [192.168.0.2:44603]
   -> 7a:27:ac:82:83:6a [192.168.0.2:60259]
   -> 7a:76:1a:fe:d2:3c [192.168.0.2:52234]
   -> 7a:71:93:51:24:98 [192.168.0.2:44247]
   -> 7a:05:6a:8c:b1:3f [192.168.0.2:55484]
   -> 7a:23:07:a5:99:a2 [192.168.0.2:51103]
Peer 7a:27:ac:82:83:6a (v711) (UID 11378769113006941369)
   -> 7a:76:1a:fe:d2:3c [192.168.0.1:6783]
   -> 7a:c3:ea:78:72:dc [192.168.4.254:6783]
   -> 7a:49:7e:1a:50:df [10.179.163.107:6783]
Peer 7a:05:6a:8c:b1:3f (v187) (UID 14782931913940871202)
   -> 7a:c3:ea:78:72:dc [10.9.168.20:6783]
   -> 7a:49:7e:1a:50:df [10.179.163.107:6783]
   -> 7a:76:1a:fe:d2:3c [192.168.0.1:6783]
Peer 7a:23:07:a5:99:a2 (v178) (UID 3201821123665204884)
   -> 7a:49:7e:1a:50:df [10.179.163.107:6783]
   -> 7a:c3:ea:78:72:dc [10.9.168.20:6783]
   -> 7a:76:1a:fe:d2:3c [10.99.185.203:6783 (unestablished)]
Peer 7a:c3:ea:78:72:dc (v98) (UID 5743069062583085255)
   -> 7a:05:6a:8c:b1:3f [192.168.4.254:56224]
   -> 7a:71:93:51:24:98 [192.168.4.254:34305]
   -> 7a:23:07:a5:99:a2 [192.168.4.254:52102]
   -> 7a:49:7e:1a:50:df [10.179.163.107:6783]
   -> 7a:76:1a:fe:d2:3c [10.99.185.203:6783]
   -> 7a:27:ac:82:83:6a [192.168.4.254:59207]
Peer 7a:76:1a:fe:d2:3c (v563) (UID 16150396005695964400)
   -> 7a:27:ac:82:83:6a [192.168.0.1:51871]
   -> 7a:49:7e:1a:50:df [10.179.163.107:6783]
   -> 7a:71:93:51:24:98 [192.168.0.1:50798]
   -> 7a:c3:ea:78:72:dc [192.168.0.1:56890]
   -> 7a:23:07:a5:99:a2 [192.168.0.1:56985 (unestablished)]
   -> 7a:05:6a:8c:b1:3f [192.168.0.1:51130]
Peer 7a:71:93:51:24:98 (v226) (UID 3222938306446983629)
   -> 7a:76:1a:fe:d2:3c [192.168.0.1:6783]
   -> 7a:49:7e:1a:50:df [10.179.163.107:6783]
   -> 7a:c3:ea:78:72:dc [10.9.168.20:6783]
Routes:
unicast:
7a:76:1a:fe:d2:3c -> 00:00:00:00:00:00
7a:05:6a:8c:b1:3f -> 7a:05:6a:8c:b1:3f
7a:27:ac:82:83:6a -> 7a:27:ac:82:83:6a
7a:49:7e:1a:50:df -> 7a:49:7e:1a:50:df
7a:71:93:51:24:98 -> 7a:71:93:51:24:98
7a:c3:ea:78:72:dc -> 7a:c3:ea:78:72:dc
7a:23:07:a5:99:a2 -> 7a:49:7e:1a:50:df
broadcast:
7a:71:93:51:24:98 -> []
7a:49:7e:1a:50:df -> []
7a:27:ac:82:83:6a -> []
7a:05:6a:8c:b1:3f -> []
7a:23:07:a5:99:a2 -> []
7a:c3:ea:78:72:dc -> []
7a:76:1a:fe:d2:3c -> [7a:23:07:a5:99:a2 7a:05:6a:8c:b1:3f 7a:27:ac:82:83:6a 7a:49:7e:1a:50:df 7a:71:93:51:24:98 7a:c3:ea:78:72:dc]
Reconnects:
10.9.168.20:6783 (next try at 2015-02-18 21:54:41.228319992 +0000 UTC)
192.168.4.254:6783 (trying since 2015-02-18 21:47:37.813818765 +0000 UTC)
192.168.4.254:52102 (trying since 2015-02-18 21:47:50.165342722 +0000 UTC)
192.168.0.2:6783 (trying since 2015-02-18 21:48:50.591836494 +0000 UTC)
192.168.4.254:56723 (trying since 2015-02-18 21:47:37.813608497 +0000 UTC)
192.168.4.254:34305 (trying since 2015-02-18 21:47:51.501319334 +0000 UTC)
192.168.4.254:56224 (trying since 2015-02-18 21:48:09.841437606 +0000 UTC)
192.168.0.2:51103 (trying since 2015-02-18 21:48:50.591827958 +0000 UTC)
```

**Comment by bboreham** [comment]
Not too sure about the root cause but "v7924" suggests that something is flapping - the version number should increment when the topology changes. 

If you can get logs and `weave status` a bit closer in time, and particularly any logs from the node(s) with high version numbers (e.g. The one on 10.179.163.107), that may help.

**Comment by bmurphy1976** [comment]
Where does the version # come from?  Is that the weave container?  I just noticed one server is running 2ccfe8038bb1 zettio/weave:0.9.0 and another server is running 5bd04c9d7f3f zettio/weave:0.9.0 so the container is clearly different.

**Comment by bmurphy1976** [comment]
Ah, ignore my previous comment.  I was looking at the container ids not the image ids.  The image ids are the same.  I'll try to get more diagnostics.

**Comment by bmurphy1976** [comment]
What qualifies as a topology change?  We have a lot of cron jobs that start short lived containers every minute and every 5 minutes.  Could that account for the high version numbers and some of the flapping?Our large volume of cron jobs cause other problems (i.e. resource leaks in docker itself) so we may be an outlier here.

Here are the last 10 minutes of logs.  First, this instance is currently exhibiting the proble:

https://gist.github.com/bmurphy1976/b512a17069fca81ec72f

This is the instance with the ip address asked for above:

https://gist.github.com/bmurphy1976/d96a35936c569f329b74

and finally this is a third instance and this is the instance that runs an excessive amount of cron jobs:

https://gist.github.com/bmurphy1976/8cde85d29f323c338ffe

**Comment by bmurphy1976** [comment]
FYI, the other servers in this cluster are very low utilization low churn servers.  These three a the busy servers.

**Comment by rade** [comment]
[Topology](http://zettio.github.io/weave/how-it-works.html#topology) has nothing to do with application containers; it's all about weave peers and the connections between them.

**Comment by bmurphy1976** [comment]
I believe I found the problem.  The weave interface on two of our servers was misconfigured.  It was configured for the 192.x.x.x/8 network when it should have been configured for the 192.168.x.x/16 network (these servers had been configured by an earlier version of our configuration scripts that had this typo).  Once I fixed this and restarted services on the affected servers the log volume went back to normal.

This looks like the cause of the problem so I'm going to close this ticket.  Thanks for the help!
